recruit/jobopenings.go

```diff
@@ -0,0 +1,456 @@
+		}
+	}
+
+	// log.Printf("%s\n", endpoint)
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return JobOpening{}, fmt.Errorf("failed to retrieve XML searched %s: %s", JobOpeningsModule, err.Error())
+	}
+
+	if v, ok := endpoint.ResponseData.(*XMLJobOpeningsResponse); ok {
+		// Convert to Json
+		xmlData := v.Result.JobOpenings.Row.FL
+		res := &data
+
+		for _, item := range xmlData {
+			switch item.Val {
+			case "Job Description":
+				data.JobDescription = item.Text
+			}
+		}
+		return *res, nil
+	}
+
+	return JobOpening{}, fmt.Errorf("no 'JobOpening' returned")
+}
+
+type XMLJobOpeningsResponse struct {
+	XMLName xml.Name `xml:"response"`
+	Text    string   `xml:",chardata"`
+	URI     string   `xml:"uri,attr"`
+	Result  struct {
+		Text        string `xml:",chardata"`
+		JobOpenings struct {
+			Text string `xml:",chardata"`
+			Row  struct {
+				Text string `xml:",chardata"`
+				No   string `xml:"no,attr"`
+				FL   []struct {
+					Text string `xml:",chardata"`
+					Val  string `xml:"val,attr"`
+				} `xml:"FL"`
+			} `xml:"row"`
+		} `xml:"JobOpenings"`
+	} `xml:"result"`
+}
+
+// https://help.zoho.com/portal/en/kb/recruit/developer-guide/api-methods/articles/getrecords#Request_Parameters
+func (c *API) XMLGetRecords(params map[string]zohoutils.Parameter) (data XMLGetRecordsResponse, err error) {
+	endpoint := zohoutils.Endpoint{
+		Name:         "jobopenings",
+		URL:          fmt.Sprintf("https://recruit.zoho.%s/recruit/private/xml/%s/getRecordById", c.ZohoTLD, JobOpeningsModule),
+		Method:       zohoutils.HTTPGet,
+		ResponseData: &XMLGetRecordsResponse{},
+		URLParameters: map[string]zohoutils.Parameter{
+			"version":       "2", // This will fetch responses based on the latest API implementation.
+			"id":            "",  // 1 - To exclude fields with "null" values while inserting data from your Recruit account.
+			"selectColumns": "",  // mandatory eg: (Job Description) | Module(Job Description)
+		},
+	}
+
+	if len(params) > 0 {
+		for k, v := range params {
+			endpoint.URLParameters[k] = v
+		}
+	}
+
+	// log.Printf("ENDPOINT: %s\n", endpoint)
+
+	err = c.Zoho.HTTPRequest(&endpoint)
+	if err != nil {
+		return XMLGetRecordsResponse{}, fmt.Errorf("failed to retrieve XML get %s: %s", JobOpeningsModule, err.Error())
+	}
+
+	if v, ok := endpoint.ResponseData.(*XMLGetRecordsResponse); ok {
+		return *v, nil
+	}
+
+	return XMLGetRecordsResponse{}, fmt.Errorf("no 'XMLGetRecordsResponse' returned")
+}
+
+type XMLGetRecordsResponse struct {
+	XMLName xml.Name `xml:"response"`
+	Text    string   `xml:",chardata"`
+	URI     string   `xml:"uri,attr"`
+	Result  struct {
+		Text        string `xml:",chardata"`
+		JobOpenings struct {
+			Text string `xml:",chardata"`
+			Row  struct {
+				Text string `xml:",chardata"`
+				No   string `xml:"no,attr"`
+				FL   []struct {
+					Text string `xml:",chardata"`
+					Val  string `xml:"val,attr"`
+				} `xml:"FL"`
+			} `xml:"row"`
+		} `xml:"JobOpenings"`
+	} `xml:"result"`
+}
```

Comment:
I don't want to implement XML at the top level package, I was so happy when Zoho migrated to a REST-like API.

Does the recruit team plan to moved these endpoints to a JSON encoding?

Reply:
I had to use XML, it was the only way to get html formatting for job description. I hate it too :) Don't know about their intention. You're free to remove it from PR.